core/src/main/java/org/apache/calcite/avatica/remote/AvaticaCommonsHttpClientImpl.java

```diff
@@ -215,7 +220,9 @@ HostnameVerifier getHostnameVerifier(HostnameVerification verification) {
         final int statusCode = response.getStatusLine().getStatusCode();
         if (HttpURLConnection.HTTP_OK == statusCode
             || HttpURLConnection.HTTP_INTERNAL_ERROR == statusCode) {
-          return EntityUtils.toByteArray(response.getEntity());
+          byte[] bytes = EntityUtils.toByteArray(response.getEntity());
```

Comment:
Is this order-dependent? I'm wondering if there's a reason you didn't change this to be:

```java
userToken = context.getUserToken();
return EntityUtils.toByteArray(response.getEntity());
```

Reply:
Good catch @joshelser 
Yes, that works too. Let me update the PR.
Somehow, I believed that the context can only be obtained after getting the entity, which is wrong.